contracts/Tendermint-ORU.sol

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.6.0 <8.0.0;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+/// @notice Submission of remote chain block header.
+struct HeaderSubmission {
+    // Block header
+    Header header;
+    // Submitter
+    address payable submitter;
+    // Ethereum block number submission was made
+    uint256 blockNumber;
+    // Simple hash of previous block's commit in ABI encoded format
+    bytes32 lastCommitHash;
+}
+
+/// @notice Remote chain block header.
+/// @dev https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#header
+struct Header {
+    uint64 height;
+    uint64 timestamp;
+    bytes32 lastBlockID;
+    bytes32 lastCommitRoot;
+    bytes32 consensusRoot;
+    bytes32 stateCommitment;
+    bytes32 availableDataRoot;
+    bytes32 proposerAddress; // Note: this needs to be converted to a 20-byte address when used
+}
+
+/// @notice Compact ECDSA signature.
+/// @dev https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#signature
+struct Signature {
+    bytes32 r;
+    bytes32 vs;
+}
+
+/// @notice Tendermint signature.
+/// @dev https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#commitsig
+struct CommitSig {
+    uint8 blockIDFlag;
+    bytes32 validatorAddress; // Note: this needs to be converted to a 20-byte address when used
+    uint64 timestamp;
+    Signature signature;
+}
+
+/// @notice Tendermint commit (list of signatures).
+/// @dev https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#commit
+struct Commit {
+    uint64 height;
+    uint64 round;
+    uint64 blockID;
+    CommitSig[] signatures;
+}
+
+/// @notice Bare minimim block data. Only contains the block header and commit.
+struct BareBlock {
```

Comment:
In the light client we often call these LightBlock.

Reply:
Will change then

Reply:
Why isn't this simply called "block header"? Because it also contains the signatures?

Reply:
Not exactly. A block contains a block header, commit sigs, an available data header, and the available data (https://github.com/lazyledger/lazyledger-specs/blob/c2caced73002d6e5b556b072ad9ca286196447fc/specs/data_structures.md#block). What I'm calling `BareBlock` has no available data. But block headers don't actually have the full Tendermint commit, they only have the root of the commit.

Reply:
Fixed in 8788251.